src/edit-flashcard.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+
+export default class EditFlashcards extends React.Component {
+  constructor(props) {
+    super(props)
+    this.saveEditedCard = this.saveEditedCard.bind(this)
+  }
+  saveEditedCard(event) {
+    event.preventDefault()
+    const uniqueId = parseInt(this.props.params.uniqueId, 10)
+    const editedCard = {
+      question: event.target.editedQuestion.value,
+      answer: event.target.editedAnswer.value,
+      id: uniqueId
+    }
+    const { flashcards } = this.props
+    const updatedFlashcards = flashcards.map(card => {
+      if (card.id === uniqueId) {
+        return editedCard
+      }
+      else {
+        return card
+      }
+    })
+    this.props.saveEditedFlashcards(updatedFlashcards)
+  }
+  render() {
+    const { flashcards, params } = this.props
+    const currentEditFlashcard = flashcards.find(card => card.id === parseInt(params.uniqueId, 10))
```

Comment:
You are making your `EditFlashcards` component do too much work to get the card it wants. Pass it only the card to edit instead of all of them.